integration/spark/app/src/test/java/io/openlineage/spark/agent/InvalidTablesInJdbcInputsTest.java

```diff
@@ -0,0 +1,261 @@
+        event -> {
+          List<OpenLineage.InputDataset> inputs = event.getInputs();
+          assertThat(inputs).hasSize(1);
+          OpenLineage.InputDataset input = inputs.get(0);
+          assertThat(input.getName()).isEqualTo("openlineage.authors");
+
+          List<SchemaRecord> schema =
+              SparkTestUtils.mapToSchemaRecord(input.getFacets().getSchema());
+
+          assertThat(schema).containsExactlyInAnyOrderElementsOf(expectedSchema);
+        });
+  }
+
+  @Test
+  void testPostgresWhenOnlyOneTableIsLoadedWithQuery() throws IOException, InterruptedException {
+    String query = loadResourceFileAsString("/queries/with_one_table.sql");
+    HttpServer server = createHttpServer(handler);
+    PostgreSQLTestContainer postgres = startPostgresContainer();
+
+    SparkSession spark =
+        createSparkSession(
+            server.getAddress().getPort(), "testPostgresWhenOnlyOneTableIsLoadedWithQuery");
+
+    loadSqlQuery(spark, postgres, query);
+
+    postgres.stop();
+    spark.stop();
+
+    List<OpenLineage.RunEvent> runEvents =
+        handler.getEvents("test_postgres_when_only_one_table_is_loaded_with_query").stream()
+            .filter(event -> !event.getInputs().isEmpty())
+            .collect(Collectors.toList());
+
+    assertThat(runEvents).hasSizeGreaterThanOrEqualTo(4);
+
+    List<SchemaRecord> expectedSchema =
+        Arrays.asList(
+            new SchemaRecord("author_name", "string"), new SchemaRecord("author_id", "integer"));
+
+    runEvents.forEach(
+        event -> {
+          List<OpenLineage.InputDataset> inputs = event.getInputs();
+          assertThat(inputs).hasSize(1);
+          OpenLineage.InputDataset input = inputs.get(0);
+          assertThat(input.getName()).isEqualTo("openlineage.authors");
+
+          List<SchemaRecord> schema =
+              SparkTestUtils.mapToSchemaRecord(input.getFacets().getSchema());
+
+          assertThat(schema).containsExactlyInAnyOrderElementsOf(expectedSchema);
+        });
+  }
+
+  @Test
+  void testPostgresQueryWhenMultipleTablesAreInTheQuery() throws IOException, InterruptedException {
+    String query = loadResourceFileAsString("/queries/with_multiple_tables.sql");
+    HttpServer server = createHttpServer(handler);
+    PostgreSQLTestContainer postgres = startPostgresContainer();
+
+    SparkSession spark =
+        createSparkSession(
+            server.getAddress().getPort(), "testPostgresQueryWhenMultipleTablesAreInTheQuery");
+
+    loadSqlQuery(spark, postgres, query);
+
+    postgres.stop();
+    spark.stop();
+
+    List<OpenLineage.RunEvent> runEvents =
+        handler.getEvents("test_postgres_query_when_multiple_tables_are_in_the_query").stream()
+            .filter(event -> !event.getInputs().isEmpty())
+            .collect(Collectors.toList());
+
+    assertThat(runEvents).hasSizeGreaterThanOrEqualTo(4);
+
+    runEvents.stream()
+        .map(OpenLineage.RunEvent::getInputs)
+        .forEach(
+            inputs -> {
```

Comment:
Are we extracting schemas in this case? If not, then it's at least worth writing a comment.

Reply:
its interesting how we can handle schema when we have more than one table, what we can do is to somehow parse the query and look for column references but I am not sure if its the case here and for simplicity is better to keep the schema to be empty StructType ?